addons/eventbus/src/main/java/org/vaadin/spring/events/internal/ScopedEventBus.java

```diff
@@ -100,17 +100,32 @@ public EventScope getScope() {
 
     @Override
     public <T> void publish(Object sender, T payload) {
-        logger.debug("Publishing payload [{}] from sender [{}] on event bus [{}]", payload, sender, this);
-        listeners.publish(new Event<T>(this, sender, payload));
+        publish("", sender, payload);
+    }
+
+    @Override
+    public <T> void publish(String target, Object sender, T payload) {
+        logger.debug("Publishing payload [{}] from sender [{}] on event bus [{}] to raget  [{}]", payload, sender, this, target);
+        listeners.publish(new Event<T>(this, sender, payload, target));
     }
 
     @Override
     public <T> void publish(EventScope scope, Object sender, T payload) throws UnsupportedOperationException {
-        logger.debug("Trying to publish payload [{}] from sender [{}] using scope [{}] on event bus [{}]", payload, sender, scope, this);
+        publish(scope, "", sender, payload);
+    }
+
+    @Override
+    public <T> void publish(EventScope scope, String target, Object sender, T payload) throws UnsupportedOperationException {
+        logger.debug("Trying to publish payload [{}] from sender [{}] using scope [{}] on event bus [{}] to target [{}]", payload, sender, scope, this, target);
+        
+        if (target == null) {
```

Comment:
Why is there a null check here, but not in the publish-method without the scope? I'd rather move the null check to the Event constructor.